### ISSUE TITLE
Instructions for the GPS Simulator App

### ISSUE THREAD

**Issue by minkione** [jamesshao8/portapack-havoc-modified#2]
Can I suggest to add somewhere the correct commands and requirements to use the gps simulator app?

Something like....
```

https://github.com/osqzss/gps-sdr-sim/wiki/Real-Time-Streaming-to-HackRF
https://slowmin.com/fake-gps-with-a-hackrf-one/
./gps-sdr-sim -e brdc3540.14n -s 2500000 -b 8 -o gpssim.c8 
./gps-sdr-sim -e brdc3540.14n -l 40,110,100 -b 8 -o gpssim.c8 
create text file named "gpssim.txt" context: 
sample_rate=2500000 center_frequency=1575420000 and copy this two files into sd card at some dir: 
./gpssim.c8 
./gpssim.txt 
```

**Comment by jamesshao8** [comment]
Thanks, I didn't write a good instruction because I am busy doing another app, analog tv receiver.
The gps instruction should be similar as yours

**Comment by minkione** [comment]
Yeah, was just a reminder for future reference for anyone that sees your repo :)
P.S. Regarding the analog receiver... if is going to be an app for portapack, let me know in case is needed some testing.
P.P.S. are you on twitter or linkedin? To keep in touch more easily than a Github issue ;]

**Comment by jamesshao8** [comment]
Hi I have a youtube channel, my account name is zyshao0@gmail.com. I am into analog video reception these days. I am planning on publishing a programming tutorial on demodulating them on Linux using HackRF/LimeSDR. And also how I did that on portapack, if I got a success. I will upload my current code soon. In my current implementation, I modified analog audio receiver (wfm mode) to show time domain signals. Changing audio fft spectrum to time domain signal, and changing channel fft waterfall to 2 dimensional AM demotulated grey scale signal (blue or green stands for white, and red stands for black). And I used https://github.com/argilo/sdr-examples/blob/master/ntsc/ntsc-encode.py , https://github.com/argilo/sdr-examples/blob/master/ntsc/ntsc_hackrf.grc to send out test signal. My test photo are 2 pictures with pure white and pure black.
You can directly find them here.
![white](https://user-images.githubusercontent.com/17997195/75167582-91093c80-5760-11ea-8dcb-32f999d5e4fc.png)
![black](https://user-images.githubusercontent.com/17997195/75167588-94042d00-5760-11ea-8e2b-a64ff72a5991.png)

Here're my results.
![white_result](https://user-images.githubusercontent.com/17997195/75167849-f8bf8780-5760-11ea-934a-0eb6ad7995e3.png)
This is the result when I am receiving white picture of ntsc signal. Between the red borders, there is a light green area. and horizontal border is narrow, while vertical border is wide, which complies with analog tv standards.
![black_result](https://user-images.githubusercontent.com/17997195/75168210-7be0dd80-5761-11ea-8d4d-2cfd2dccc461.png)
This is the result when I am receiving black picture. The area between red borders is also red, stands for darker color.

If you'd like to test that, remember to adjust the lna and vga on portapack so that you will see clear bondary in red color, and some green color between them.

**Comment by antonlalala** [comment]
Link to you channel on ytb

**Comment by minkione** [comment]
> Link to you channel on ytb

How cute... a lurker popped up :D :D :D 🤣

**Comment by jamesshao8** [comment]
https://www.youtube.com/channel/UCbxa-MHpncOnZHUWtl9YcIg/
@minkione  @antonlalala

**Comment by jamesshao8** [comment]
https://www.youtube.com/watch?v=1RzKzRdeTiQ
I have uploaded a video. I am using portapack to demodulate pal video from my quadcopter's camera. You can see there is a curve like a snake between the wide border on portapack's screen, and that curve is changing according to my hand's movement. @minkione

**Comment by minkione** [comment]
Nice!
Wonder what comrad Anton thinks about it 😉

**Comment by jamesshao8** [comment]
@minkione I have already finished analog video demodulation. It's working quite well for PAL signal. For ntsc, it also works, but the parameters are not optimized, so not as good as pal. I have uploaded my latest code you can have a try there.

demo video:
https://youtu.be/VxBReLPbfZc

**Comment by minkione** [comment]
Will try as soon as I find a PAL signal

**Comment by jamesshao8** [comment]
You can use hacktv as signal source. On ubuntu 18.04, it's easy to install.
It can transmit a video file on your computer using hackrf. both pal and ntsc are supported
https://github.com/fsphil/hacktv

**Comment by jamesshao8** [comment]
@minkione  how is your testing with analog tv?
I made a new app of bluetooth receiving. you can check it out if interested.

**Comment by michaelbuud19** [comment]
hi
do you have bin file for portapack  ?
i love to give a test

**Comment by eried** [comment]
@michaelbuud19 hey Michael, can you share a file to test the GPS sim? I am trying to port this awesome Shao apps to the official havoc so they appear on the next official release, but I want to test them and write some instructions

**Comment by jamesshao8** [comment]
Yesterday, I was trying to upload the gps file to dropbox, but I failed. It's very difficult for me to upload big files to foreign server. Maybe @minkione can help you with that.

**Comment by eried** [comment]
Thanks, I created one with your instructions (40,110 as lat,long) but my antena is just the included with the portapack h2 from aliexpress so my smartwatch nor my phone receive the "tampered" gps signal. Is it normal that the progress bar does not advance in the gps replayer?

**Comment by eried** [comment]
Thanks, I created one with your instructions (40,110 as lat,long) but my antena is just the included with the portapack h2 from aliexpress so my smartwatch nor my phone receive the "tampered" gps signal. Is it normal that the progress bar does not advance in the gps replayer?

**Comment by jamesshao8** [comment]
It is normal that progress bar doesnt advance.
What kind of antenna you are using? The signal doesnt need to be strong, in fact receiver can't lock if signal is too strong.
Are you using an android phone or iphone? If it's android, you can install an app called gps test to see the signal in detail.
For some phone, the time in gps file is also important. You need to specify it by parameter "-t", when creating the iq file.

**Comment by michaelbuud19** [comment]
> Yesterday, I was trying to upload the gps file to dropbox, but I failed. It's very difficult for me to upload big files to foreign server. Maybe @minkione can help you with that.

 I manage to download the file from Baidu you share  . thanks
I se you have SD card with GPS bin  file include
